package/kernel/mac80211/patches/rt2x00/602-01-wifi-rt2x00-Add-support-for-loading-EEPROM-from-user.patch

```diff
@@ -233,57 +175,18 @@
 +		goto err_release_ee;
 +	}
 +
-+	rt2x00dev->eeprom_file = ee;
-+	return 0;
++	memcpy(rt2x00dev->eeprom, ee->data, rt2x00dev->ops->eeprom_size);
 +
 +err_release_ee:
 +	release_firmware(ee);
 +err_exit:
 +	return retval;
 +}
 +
-+int rt2x00lib_load_eeprom_file(struct rt2x00_dev *rt2x00dev)
-+{
-+	int retval;
-+
-+	retval = rt2x00lib_request_eeprom_file(rt2x00dev);
-+	if (retval)
-+		return retval;
-+
-+	return 0;
-+}
-+
-+void rt2x00lib_free_eeprom_file(struct rt2x00_dev *rt2x00dev)
-+{
-+	if (rt2x00dev->eeprom_file && rt2x00dev->eeprom_file->size)
-+		release_firmware(rt2x00dev->eeprom_file);
-+	rt2x00dev->eeprom_file = NULL;
-+}
---- a/drivers/net/wireless/ralink/rt2x00/rt2x00lib.h
-+++ b/drivers/net/wireless/ralink/rt2x00/rt2x00lib.h
-@@ -286,6 +286,22 @@ static inline void rt2x00lib_free_firmwa
- #endif /* CPTCFG_RT2X00_LIB_FIRMWARE */
- 
- /*
-+ * EEPROM file handlers.
-+ */
-+#ifdef CPTCFG_RT2X00_LIB_EEPROM
-+int rt2x00lib_load_eeprom_file(struct rt2x00_dev *rt2x00dev);
-+void rt2x00lib_free_eeprom_file(struct rt2x00_dev *rt2x00dev);
-+#else
-+static inline int rt2x00lib_load_eeprom_file(struct rt2x00_dev *rt2x00dev)
++int rt2x00lib_read_eeprom(struct rt2x00_dev *rt2x00dev)
```

Comment:
We need to export this function to avoid build error.`EXPORT_SYMBOL_GPL(rt2x00lib_read_eeprom);`

Reply:
You had to edit this yourself? exporting the symbol is not needed. Can you post error you have?

Reply:
```
Building backport-include/backport/autoconf.h ... done.
  MODPOST /home/db/openwrt/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7620/backports-6.5/Module.symvers
ERROR: modpost: "rt2x00lib_read_eeprom" [/home/db/openwrt/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7620/backports-6.5/drivers/net/wireless/ralink/rt2x00/rt2800soc.ko] undefined!
make[7]: *** [scripts/Makefile.modpost:133: /home/db/openwrt/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7620/backports-6.5/Module.symvers] Error 1
make[7]: *** Deleting file '/home/db/openwrt/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7620/backports-6.5/Module.symvers'
make[6]: *** [Makefile:1827: modules] Error 2
make[5]: *** [Makefile.build:13: modules] Error 2
make[4]: *** [Makefile.real:93: modules] Error 2
make[3]: *** [Makefile:121: modules] Error 2
make[3]: Leaving directory '/home/db/openwrt/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7620/backports-6.5'
make[2]: *** [Makefile:401: /home/db/openwrt/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7620/backports-6.5/.built] Error 2
make[2]: Leaving directory '/home/db/openwrt/package/kernel/mac80211'
time: package/kernel/mac80211/regular/compile#1.17#0.34#1.87
    ERROR: package/kernel/mac80211 failed to build (build variant: regular).
make[1]: *** [package/Makefile:120: package/kernel/mac80211/compile] Error 1
make[1]: Leaving directory '/home/db/openwrt'
make: *** [/home/db/openwrt/include/toplevel.mk:232: package/kernel/mac80211/compile] Error 2
```

Reply:
They really love to create kmods my god....